### PR TITLE
Upgrade to v15

### PR DESCRIPTION
**Pull request by jamiebrynes7**
This PR contains the content from #1491 only related to updating the GDK to v15. In a follow-up PR, I will add the load balancing feature module.

Note that if you run this code as it stands, nothing happens... because there's no load balancing happening.

### CODE REVIEW

**Review by zeroZshadow — APPROVED** [review]

File: workers/unity/Packages/io.improbable.gdk.core/.codegen/Source/Generators/Core/UnityEcsViewManagerGenerator.cs [hunk 1]
```diff
@@ -13,12 +13,14 @@ public static CodeWriter Generate(UnityComponentDetails componentDetails)
         {
             var componentNamespace = $"global::{componentDetails.Namespace}.{componentDetails.Name}";
 
+            var nonBlittableFields = componentDetails.FieldDetails
+                .Where(fd => !fd.IsBlittable)
+                .ToList();
+
+
```
Comment (paulbalaji) [inline workers/unity/Packages/io.improbable.gdk.core/.codegen/Source/Generators/Core/UnityEcsViewManagerGenerator.cs:20]: newline

File: workers/unity/Assets/Playground/Editor/SnapshotGenerator/SnapshotGenerator.cs [hunk 2]
```diff
@@ -55,6 +57,7 @@ private static void CreateSpinner(Snapshot snapshot, Coordinates coords)
             snapshot.AddEntity(template);
         }
 
+
```
Comment (paulbalaji) [inline workers/unity/Assets/Playground/Editor/SnapshotGenerator/SnapshotGenerator.cs:60]: newline

File: workers/unity/Assets/Playground/Scripts/Worker/ClientWorkerConnector.cs [hunk 3]
```diff
@@ -1,12 +1,18 @@
...
+using Improbable.Gdk.PlayerLifecycle;
+using Improbable.Gdk.TransformSynchronization;
+using Improbable.Worker.CInterop;
 using UnityEngine;
 
 namespace Playground
 {
     public class ClientWorkerConnector : WorkerConnector
     {
+        public const string UnityClient = "UnityClient";
```
Comment (paulbalaji) [inline workers/unity/Assets/Playground/Scripts/Worker/ClientWorkerConnector.cs:14]: rename to `WorkerType` in here + other connectors?

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,19 +2,46 @@
...
 ### Breaking Changes
 
+- `CallerWorkerId` and `CallerAttributeSet` have been replaced with `CallerWorkerEntityId` on all received command requests.
+- `EntityQuery` no longer supports the `Count` result type.
+- `ModularKcp` and `ModularTcp` have been renamed to `Kcp` and `Tcp`. `RakNet` and the previous `Tcp` implementations have been removed as available network connection types.
+- Read and write acl have been removed from the `EntityTemplate` class & the helper methods for adding feature module components to an entity:
+  - `EntityTemplate.GetWorkerAccessAttribute()`, `EntityTemplate.GetComponentWriteAccess()`, `EntityTemplate.SetComponentWriteAccess()`, `EntityTemplate.SetComponentWriteAccess()`, `EntityTemplate.SetReadAccess()` have all been removed.
+  - The remaining function signatures on the `EntityTemplate` class have been adjusted to remove references to ACL.
+  - The function signatures on the feature module helper functions (e.g., `TransformSynchronizationHelper.AddTransformSynchronizationComponents`) have also been adjusted.
+- `CredentialsType` has been removed from the `LocatorParamters` class.
```
Comment (paulbalaji) [inline CHANGELOG.md:16]: `LocatorParameters`

**Review by paulbalaji — APPROVED** [review]
generally lgtm

File: workers/unity/Assets/Playground/Scripts/Worker/ClientWorkerConnector.cs (hunk 3 above)
Comment (jamiebrynes7) [reply workers/unity/Assets/Playground/Scripts/Worker/ClientWorkerConnector.cs:14]: See comment [here](https://github.com/spatialos/gdk-for-unity/pull/1491#discussion_r560961852) 😄

Comment (paulbalaji) [reply workers/unity/Assets/Playground/Scripts/Worker/ClientWorkerConnector.cs:14]: 👍

File: init.ps1 [hunk 5]
```diff
@@ -52,14 +47,11 @@ UpdatePackage schema test_schema_library "schema"
...
-UpdateSpot spot-win64 "$SdkPath/.spot/spot.exe"
-UpdateSpot spot-macos "$SdkPath/.spot/spot"
-
 #Update Mobile SDK
 UpdatePackage worker_sdk c-static-arm-clang-ios "$SdkMobilePath/Plugins/Improbable/Core/iOS/arm"
 UpdatePackage worker_sdk c-static-x86_64-clang-ios "$SdkMobilePath/Plugins/Improbable/Core/iOS/x86_64"
 
-UpdatePackage worker_sdk c-dynamic-arm64v8a-clang_ndk21-android "$SdkMobilePath/Plugins/Improbable/Core/Android/arm64"
-UpdatePackage worker_sdk c-dynamic-armv7a-clang_ndk21-android "$SdkMobilePath/Plugins/Improbable/Core/Android/armv7"
+UpdatePackage worker_sdk c-dynamic-arm64v8a-clang_ndk21d-android "$SdkMobilePath/Plugins/Improbable/Core/Android/arm64"
```
Comment (austinkelmore) [inline init.ps1:54]: Does this change mean the debug packages are used? Is this intended?

File: workers/unity/Assets/Tests/EditmodeTests/Correctness/PlayerLifecycle/PlayerLifecycleHelperTests.cs [hunk 6]
```diff
@@ -16,6 +15,7 @@ public class PlayerLifecycleHelperTests
         [SetUp]
         public void Setup()
         {
+            // TODO: The WorkerEntityId for this worker is '0'. Probably should refactor this in the future.
```
Comment (austinkelmore) [inline workers/unity/Assets/Tests/EditmodeTests/Correctness/PlayerLifecycle/PlayerLifecycleHelperTests.cs:18]: Worth doing now, or is this a future thing?

File: init.ps1 (hunk 5 above)
Comment (jamiebrynes7) [reply init.ps1:54]: Nope, NDK 21d is the LTS version of NDK 21 (https://developer.android.com/ndk/downloads).

I'm unclear why the revisions are named as such.

File: workers/unity/Assets/Tests/EditmodeTests/Correctness/PlayerLifecycle/PlayerLifecycleHelperTests.cs (hunk 6 above)
Comment (jamiebrynes7) [reply workers/unity/Assets/Tests/EditmodeTests/Correctness/PlayerLifecycle/PlayerLifecycleHelperTests.cs:18]: Not impacting anything now, so no need right now. If we run into a case where we need to do it, we can tweak it

**Pull request merged**